### ISSUE TITLE
Selecting sleep when done no longer works

### ISSUE THREAD

**Issue by 4-testing** [AaronFeng753/Waifu2x-Extension-GUI#286]
Selecting sleep when done no longer works. I get a box asking whether I want to sleep immediately, and it goes no further. I want to leave a project running and have the computer sleep when done automatically This no longer works..

**Comment by 4-testing** [comment]
Using latest beta version .10

**Comment by AaronFeng753** [comment]
What do you mean by no longer works, it won't go to sleep after 60 seconds?

**Comment by 4-testing** [comment]
Yes. It just sits at the question screen. The next morning it still hadn't
shut down

**Comment by 4-testing** [comment]
I am running Win 7

**Comment by AaronFeng753** [comment]
I just tested it. Works fine on my windows 10 laptop. Maybe nircmd has some compatibility issues with windows 7? Are you sure this go to sleep thing works fine on your win7 pc before?(I mean the previous version) because I didn't change the method of make pc go to sleep.

**Comment by 4-testing** [comment]
It worked. Up until the latest beta version.

**Comment by AaronFeng753** [comment]
Could you restart your PC and do a quick test if it still doesn't work? I didn't saw any problems in the code, it was always just call nircmd to make pc go to sleep, I never change that after it's implemented

**Comment by AaronFeng753** [comment]
But I did find another bug related to that pop up window, but I am not sure if it will affect the sleep

**Comment by 4-testing** [comment]
Ok, I'll check it out.

**Comment by AaronFeng753** [comment]
Thanks

**Comment by 4-testing** [comment]
I just remembered something. I have the ram-disk enabled. It speeds up
processing on photos too. I am retesting and will let you know.

**Comment by AaronFeng753** [comment]
I just did a test with ram disk enabled, doesn't make any difference, PC still gose to sleep like normal

**Comment by 4-testing** [comment]
I just ran it with ram-disk disabled and it worked. I will test with it
enabled again

**Comment by AaronFeng753** [comment]
Okay, so the sleep function works.
I don't think the ram disk can stop pc go to sleep, even if it can, I cannot do anything about it, most of the ram disk operations are handled by imdisk toolkit

**Comment by 4-testing** [comment]
Some new information. It will sleep after 60 seconds, but this is from the
time the screen is activated. If the screen is deactivated, such as with
your button or timeout. it will not go to sleep until 60 seconds after the
screen comes on.

**Comment by AaronFeng753** [comment]
Okay, this is  definitely windows' problem, I just tested this on my win10 laptop and everything works fine, I have a browser on the front and PC still goes to sleep.
I saw this problem before on my windows 2 in 1, which is running an old version win10, my GUI will freeze when screen is locked. But I never encounter this problem on my other 2 laptops(which are running the latest ver of win10)

**Comment by AaronFeng753** [comment]
> Okay, this is definitely windows' problem, I just tested this on my win10 laptop and everything works fine, I have a browser on the front and PC still goes to sleep.
> I saw this problem before on my windows 2 in 1, which is running an old version win10, my GUI will freeze when screen is locked(because the GUI is not actively selected, that's why there is a turn off screen button on my GUI). But I never encounter this problem on my other 2 laptops(which are running the latest ver of win10)

So I cannot fix this, because I don't know what Windows did to my GUI, and a system update could fix it anyway

**Comment by 4-testing** [comment]
I finally figured it out,At the time the message comes up, there is a popup
asking for a donation. The 60 sec. doesn't start until that window is
cleared. This defeats the purpose of an automatic shutdown. This screen
should be placed after restart.

**Comment by AaronFeng753** [comment]
Yeah, I never tested this on free version before (obviously I always use premium version)
And yes, it is a bug, I will fix this

**Comment by AaronFeng753** [comment]
It's fixed now, this fix will be included in the next update

**Comment by 54724531** [comment]
I want to support you @AaronFeng753 but I am unable to pay you every month...if I donate do I get to keep the version even when my subscription runs out, just no updates? how does that work?  I've scoured your github all over to no avail...thank you

**Comment by AaronFeng753** [comment]
Here is how subscription works on patreon:

Depends on your tier, you can get 1 or more activation codes at the first month, since you only wanna pay me once, you can download the latest available version,  and ask me for activation code before your subscription expires.

And you can keep using that premium copy even after your subscription expired, but you can not get more premium updates.

If you only wanna pay me once but still wanna get updates, you can choose choose the annual subscription on patreon, you can receive updates for 1 year and get more activation codes

(I don't recommend you to use PayPal, right now it's for donations only)

@54724531

**Comment by AaronFeng753** [comment]
Since this is a reallllly old issue, I will lock it, if you have more questions, please contact me on my patreon page, this is GitHub issue section, I don't wanna talk too much about how subscription works at here

@54724531
